src/main/java/seedu/duke/storage/GetFromTxt.java

```diff
@@ -0,0 +1,118 @@
+import java.util.ArrayList;
+import java.util.Scanner;
+
+/**
+ * The GetFromTxt class is responsible for reading financial records data from a text file.
+ * It parses the contents of the file and populates two ArrayLists: one for Incomes and one for Expenses.
+ *
+ * This class handles exceptions for file creation errors, parsing errors, and other issues that may arise
+ * when reading data from the file.
+ */
+public class GetFromTxt {
+    private static final String AMOUNT_NOT_SUPPORT_ERROR =
+            "Amount field from this line is not supported from storage file ";
+
+    private static final String DATE_TIME_FORMAT_ERROR =
+            "Date time format incorrect on this line from storage file ";
+
+    private static final String FILE_CREATION_ERROR =
+            "Something went wrong while creating the file from storage file ";
+
+    private static final String INDEX_OUT_OF_BOUND_FROM_FILE_ERROR =
+            "Details missed on this line from storage file ";
+
+    private static final String NUMBER_FROM_FILE_INCORRECT =
+            "The expense type is incorrect on this line from storage file ";
+
+    private final File file;
+
+    /**
+     * Construct a new object of the GetFromTxt class with the provided file path.
+     *
+     * @param path The path to the text file containing financial records data.
+     */
+    public GetFromTxt(String path) {
+        file = new File(path);
+    }
+
+    /**
+     * Reads financial records data from the text file and add to the corresponding ArrayLists
+     * with Income and Expense objects.
+     *
+     * @param incomes   The ArrayList to store Income objects.
+     * @param expenses  The ArrayList to store Expense objects.
+     * @throws FileNotFoundException If the file specified by the path cannot be found.
+     */
+    public void getFromTextFile(ArrayList<Income> incomes, ArrayList<Expense> expenses) throws FileNotFoundException {
+        try {
+            if (file.createNewFile()) {
+                return; // If there is no such file in the directory, create the file and return without reading data.
+            }
+        } catch (IOException e) {
+            System.out.println(FILE_CREATION_ERROR);
+        }
+        Scanner s = new Scanner(file);                      // Create a Scanner using the File as the source
+        while (s.hasNext()) {
+            String textLine = s.nextLine();
+            try {
+                // Parse the data from the text line
+                int expenseType = 0;
+                String commandFromFile = textLine.split(" \\| ")[0];
+                String descriptionFromFile = textLine.split(" \\| ")[1];
+                double amountFromFile = Double. parseDouble(textLine.split(" \\| ")[2]);
+                LocalDate dateFromFile =  LocalDate.parse(textLine.split(" \\| ")[3]);
+                // Check for valid amount range
+                if (amountFromFile > 999999.99 || amountFromFile <= 0) {
```

Comment:
Perhaps you could convert this into a constant variable rather than a string literal in case you'd like to change your boundary values in the future and it needs to be used elsewhere in your codebase.